From ba33af7634f217f0a7a838754e1e542314dd51d5 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Sat, 14 May 2022 22:57:39 +0800
Subject: [PATCH 24/58] Revert "drm/amdkfd: Create file descriptor after client
 is added to smi_clients list"

This reverts commit 82e43950143cc92534c359f5fa42092c928930e7.
---
 drivers/gpu/drm/amd/amdkfd/kfd_smi_events.c | 24 ++++++++-------------
 1 file changed, 9 insertions(+), 15 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_smi_events.c b/drivers/gpu/drm/amd/amdkfd/kfd_smi_events.c
index bd4caa36ab2e2..17d1736367ea3 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_smi_events.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_smi_events.c
@@ -270,29 +270,23 @@ int kfd_smi_event_open(struct kfd_dev *dev, uint32_t *fd)
 		return ret;
 	}
 
-	init_waitqueue_head(&client->wait_queue);
-	spin_lock_init(&client->lock);
-	client->events = 0;
-	client->dev = dev;
-
-	spin_lock(&dev->smi_lock);
-	list_add_rcu(&client->list, &dev->smi_clients);
-	spin_unlock(&dev->smi_lock);
-
 	ret = anon_inode_getfd(kfd_smi_name, &kfd_smi_ev_fops, (void *)client,
 			       O_RDWR);
 	if (ret < 0) {
-		spin_lock(&dev->smi_lock);
-		list_del_rcu(&client->list);
-		spin_unlock(&dev->smi_lock);
-
-		synchronize_rcu();
-
 		kfifo_free(&client->fifo);
 		kfree(client);
 		return ret;
 	}
 	*fd = ret;
 
+	init_waitqueue_head(&client->wait_queue);
+	spin_lock_init(&client->lock);
+	client->events = 0;
+	client->dev = dev;
+
+	spin_lock(&dev->smi_lock);
+	list_add_rcu(&client->list, &dev->smi_clients);
+	spin_unlock(&dev->smi_lock);
+
 	return 0;
 }
-- 
2.27.0

